### PR TITLE
Add pagination, sorting and filtering to thirdpartyrepo endpoint

### PR DESCRIPTION
**Pull request by Saurabh-Thakre**

### CODE REVIEW

**Comment by Saurabh-Thakre** [comment]
Adding docs in some time

File: pkg/routes/thirdpartyrepo.go [hunk 1]
```diff
@@ -108,15 +128,34 @@ func GetAllThirdPartyRepo(w http.ResponseWriter, r *http.Request) {
...
+	log.Debugf("r.URL.Query() %v \n", r.URL.Query().Get("sort_by"))
+	if r.URL.Query().Get("sort_by") != "name" && r.URL.Query().Get("sort_by") != "-name" {
+		result = result.Limit(pagination.Limit).Offset(pagination.Offset).Where("account = ?", account).Find(&tprepo)
+	}
+	filter := r.URL.Query().Get("filter")
+	filterMap := map[string]string{}
+	if filter != "" {
+		filterMap, err = validateFilterMap(filter)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (biancarosa) [inline pkg/routes/thirdpartyrepo.go:148]: Can we use the errors.NewBadRequest() wrapper to return in the correct API contract?

File: pkg/routes/thirdpartyrepo.go [hunk 2]
```diff
@@ -108,15 +128,34 @@ func GetAllThirdPartyRepo(w http.ResponseWriter, r *http.Request) {
...
+			return
+		}
+	}
+	if err := result.Where(filterMap).Limit(pagination.Limit).Offset(pagination.Offset).Find(&tprepo).Error; err != nil {
+		fmt.Println(err)
+		http.Error(w, "this is not a valid filter. filter must be in name.parameter", http.StatusInternalServerError)
```
Comment (biancarosa) [inline pkg/routes/thirdpartyrepo.go:154]: Can we use the errors.NewBadRequest() wrapper to return in the correct API contract? And not use fmt.Println?

**Review by biancarosa — CHANGES_REQUESTED** [review]

**Review by biancarosa — APPROVED** [review]
LGTM

**Pull request merged**